db/wal_manager.cc

```diff
@@ -507,10 +507,18 @@ Status WalManager::ReadFirstLine(const std::string& fname,
     }
   }
 
-  // ReadRecord might have returned false on EOF, which means that the log file
-  // is empty. Or, a failure may have occurred while processing the first entry.
-  // In any case, return status and set sequence number to 0.
-  *sequence = 0;
+  if (status.ok() && reader.IsCompressedAndEmptyFile()) {
+    // In case of wal_compression, it writes a `kSetCompressionType` record
+    // which is not associated with any sequence number. As result for an empty
+    // file, GetSortedWalsOfType() will skip these WALs causing the operations
+    // to fail.
```

Comment:
> As result for an empty file, GetSortedWalsOfType() will skip these WALs causing the operations to fail.

With this PR's fix, is this still true? No operation will fail, right?

Reply:
I will rewrite the comment to mention operations will not fail.

Reply:
An empty file skipped by `GetSortedWalsOfType()` should not fail because, even if we enable WAL tracking in manifest, we do not log a WAL of sync size 0 to the MANIFEST. Therefore, when we recover later, we won't be looking for such WALs.